src/reducer.js

```diff
@@ -38,24 +40,72 @@ const pathToArr = path => path.split(/\//).filter(p => !!p)
  * @return {Map} State
  */
 export default (state = initialState, action = {}) => {
-  const { path } = action
+  const { path, timestamp, requesting, requested } = action
   let pathArr
+  let rootPathArr
   let retVal
 
   switch (action.type) {
 
+    case START:
+      pathArr = pathToArr(path)
+      retVal = (requesting !== undefined)
+          ? state.setIn(['requesting', ...pathArr], fromJS(requesting))
+          : state.deleteIn(['requesting', ...pathArr])
+
+      retVal = (requested !== undefined)
+          ? retVal.setIn(['requested', ...pathArr], fromJS(requested))
+          : retVal.deleteIn(['requested', ...pathArr])
+
+      return retVal;
+
     case SET:
-      const { data } = action
+      const { data, rootPath } = action
       pathArr = pathToArr(path)
+      rootPathArr = pathToArr(rootPath)
       retVal = (data !== undefined)
         ? state.setIn(['data', ...pathArr], fromJS(data))
         : state.deleteIn(['data', ...pathArr])
 
+      retVal = (timestamp !== undefined)
+        ? retVal.setIn(['timestamp', ...rootPathArr], fromJS(timestamp))
```

Comment:
if you can't think about any case that will be delay between the dispatch and the time it will get to the reducer to this line so its ok, but if some delay will occur like on the prev line of setting the data on state (big data maybe ?) the times will not be the same.
its up to you :) for me i always prefer that the reducer will not have control on values like dates and it will come from outside.